### PR TITLE
Add discoverd registry

### PR DESCRIPTION
**Pull request by lmars**

### CODE REVIEW

**Comment by lmars** [comment]
Tested manually as part of booting Flynn and it works :smile:

/cc @titanous.

File: discoverd/registry.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+	"github.com/flynn/flannel/Godeps/_workspace/src/github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flannel/subnet"
+)
+
+// Network is stored in service metadata and contains both the config and list
+// of subnet leases.
+type Network struct {
+	Config  json.RawMessage   `json:"config"`
+	Subnets map[string][]byte `json:"subnets"`
```
Comment (titanous) [inline discoverd/registry.go]: This should be a `map[string]*json.RawMessage` otherwise it will be encoded using base64.

**Comment by lmars** [comment]
@titanous comments addressed

File: discoverd/registry.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+// WatchSubnets waits for a service metadata event with an index greater than
+// a given value, and then returns a response.
+func (r *registry) WatchSubnets(since uint64, stop chan bool) (*subnet.Response, error) {
+	for {
+		select {
+		case event, ok := <-r.events:
+			if !ok {
+				return nil, errors.New("unexpected close of discoverd event stream")
+			}
+			if event.Kind == discoverd.EventKindServiceMeta {
```
Comment (titanous) [inline discoverd/registry.go]: Could invert this (if !=; continue)

**Comment by titanous** [comment]
LGTM pending minor comment.

**Pull request merged**